docs/versioned_docs/version-zenith/guides/620942-github-google-cloud.mdx

```diff
@@ -0,0 +1,329 @@
+      - It invokes the `build()` function to produce a `Container` representing the application image.
+      - It uses the `Container.publish()` method to publish the container to the Google Artifact Registry repository and returns the SHA identifier of the published image.
+
+    The `deploy()` function takes care of deploying the application to an existing Google Cloud Run service.
+      - It accepts six arguments: strings representing the Google Cloud project name, Google Artifact Registry location, Google Artifact Registry repository name, Google Cloud Run service location, Google Cloud Run service name, and a `Secret` representing Google Cloud credentials.
+      - It invokes the `publish()` function to build and publish a `Container` representing the application image.
+      - It uses the Google Cloud Run module to deploy the container image to the named Google Cloud Run service.
+
+</TabItem>
+</Tabs>
+
+## Step 6: Test the Dagger pipeline on the local host
+
+<PartialGoogleCloudSdkCredentialsSetup />
+
+Once credentials are configured, test the Dagger Functions by running the command below from the application directory. Remember to update the `registry-location`, `repository`, `service`, and `service-location` arguments if you specified different values when creating the various Google Cloud services in Step 3 and Step 4.
+
+```shell
+dagger -m ci call --source . deploy --project myproject --registry-location us-central1 --repository myregistry/myapp --service-location us-central1 --service myservice  --credential env:GOOGLE_CREDENTIALS
+```
+
+Dagger performs the operations defined in the Dagger Function, logging each operation to the console. At the end of the process, the built container is deployed to Google Cloud Run and the function returns the deployment URL. Browse to the deployment URL to see the running application.
+
+If you deployed the example application from [Appendix A](#appendix-a-create-a-github-repository-with-an-example-express-application), you should see a page similar to that shown below:
+
+![Result of running pipeline from local host](/img/current_docs/guides/github-google-cloud/local-deployment.png)
+
+## Step 7: Create a GitHub Actions workflow
+
+Dagger executes Dagger Functions in standard OCI containers. This means that a Dagger Function will always run the same way, whether on on your local machine or a remote server.
+
+This also means that it's very easy to move the workflow you just encapsulated in Dagger Functions from your local host to GitHub Actions - all that's needed is to define a GitHub Actions workflow to call your Dagger Functions on every commit.
+
+1. Commit and push the Dagger Module to the application's GitHub repository:
+
+    ```shell
+    git add .
+    git commit -a -m "Added module"
+    git push
+    ```
+
+1. In the GitHub repository, create a new workflow file at `.github/workflows/main.yml` with the following content:
+
+    ```yaml file=./snippets/github-google-cloud/main.yml
```

Comment:
Weird that in preview the syntax highlighter only highlights the string. Everything else is white.